docs/concepts/executor/dynamic_batching.md

```diff
@@ -0,0 +1,123 @@
+(executor-dynamic-batching)=
+# Dynamic Batching
+Dynamic batching is a feature of Jina that allows requests to be accumulated and batched together before being sent to 
+an {class}`~jina.Executor`. The batch is created dynamically depending on the per-endpoint dynamic batching configuration.
+
+This feature is relevant especially for inference tasks where model inference is more optimized when done in batches in 
+order to efficiently use GPU resources.
+
+## Overview
+Enabling dynamic batching on Executor endpoints that perform inference typically results in increased throughput. 
```

Comment:
This is not true, it only happens in some circumstances. I think we should mention better hardware usage